main.py

```diff
@@ -0,0 +1,105 @@
+from fastapi import FastAPI, Depends, HTTPException
+from sqlalchemy.orm import Session
+
+import crud
+import schemas
+from db.database import SessionLocal
+
+app = FastAPI()
+
+
+def get_db() -> Session:
+    db = SessionLocal()
+    try:
+        yield db
+    finally:
+        db.close()
+
+
+@app.get("/author/", response_model=list[schemas.Author])
+def read_authors(skip: int = 0, limit: int = 5, db: Session = Depends(get_db)):
+    authors = crud.get_all_authors(db, skip=skip, limit=limit)
+    return authors
+
+
+@app.get("/author/{author_id}/", response_model=schemas.Author)
+def read_single_author(author_id: int, db: Session = Depends(get_db)):
+    db_book = crud.get_author(db=db, author_id=author_id)
+
+    if db_book is None:
+        raise HTTPException(status_code=404, detail="Author not found")
+
+    return db_book
+
+
+@app.post("/author/", response_model=schemas.Author)
+def create_author(
+    author: schemas.AuthorCreate,
+    db: Session = Depends(get_db),
+):
+    db_author = crud.get_author_by_name(db=db, name=author.name)
+
+    if db_author:
+        raise HTTPException(
+            status_code=400,
+            detail="Such name for author already exists"
+        )
```

Comment:
now you know that such validation never appears at endpoints:)
you may try to fix it, but i will not reject because of it. for now..